source/Cosmos.Core/IOPort.cs

```diff
@@ -52,7 +15,7 @@ protected IOPortBase(ushort aBase, ushort aOffset)
         /// <param name="aPort">A port to write to.</param>
         /// <param name="aData">A data.</param>
         [PlugMethod(PlugRequired = true)]
-        static protected void Write8(ushort aPort, byte aData) => throw null;
+        public static void Write8(ushort aPort, byte aData) => throw null;
```

Comment:
The number of I/O ports, on most machines, never exceeds ushort.MaxValue. In fact, x86 even limits you to that. However I don't see how making everything int instead of ushort will save us casting.

Reply:
Internally in .Net almost everything happens using int32 and anytime a calculation requires something else we need to convert it to that format. This PR itself contains quite a lot of casting to ushorts. Invalid ports can still happen with uint16 so I dont think that they lead to better code but come at a small code quality and runtime cost.

Reply:
Alright, I'll try to convert as much to Int32 as possible!